### DISCUSSION TITLE
ErrorBoundary with Scripts causes fetch with _data param, re-renders with empty statusText

### DISCUSSION THREAD

**Discussion by gar1t** [remix-run/remix#8460] [Q&A]
I'm throwing a 404 error from a loader and rendering it with an error boundary defined in `root.tsx`. I'm including `<Scripts />` in the `body` of the error boundary page.

Here's the thrown error from a route loader:

```typescript
export async function loader() {
  throw new Response(null, {
    status: 404,
    statusText: "Not Found",
  });
}
```

Here's the error boundary in `root.tsx`:

```typescript
export function ErrorBoundary() {
  const error = useRouteError();
  return (
    <html>
      <body>
        {error.statusText}
        <Scripts />
      </body>
    </html>
  );
}
```

With `<Scripts />` in the error page, I see a second call -- a `FETCH` -- to the route, but with a `?_data=<encoded-route-path>`.

My first question is, what's going on here?

My second question relates to deployment as a Cloudflare worker. When I run locally with Miniflare, the value of `error.statusText` remains unchanged from the server-side rendering to whatever's going with the fetch. In production though, the value for `error.statusText` on the fetch is blank (null/undefined/empty-string) and so I get a flash: I first see the expected/correct error text and then it sadly disappears.

I'm addressing this by not including `<Scripts />` in the error boundary, but this leaves me puzzled on the two fronts: 

1. What's going on with the fetch with `_data` in the first place?
2. Why might the behavior be different in a production setting when run as a Cloudflare worker?

**Comment by meglio** [comment]
Same error for me: statusText renders as non-empty for a moment, then becomes empty.
